src/tools/filters/applyFilter.ts

```diff
@@ -0,0 +1,176 @@
+// minchev 70 normal a
+function applySepia(data: Uint8ClampedArray, intensity: number): void {
+    for (let i = 0; i < data.length; i += 4) {
+        const r = data[i];
+        const g = data[i + 1];
+        const b = data[i + 2];
+
+        const newR = Math.min(255, 0.393 * r + 0.769 * g + 0.189 * b);
+        const newG = Math.min(255, 0.349 * r + 0.686 * g + 0.168 * b);
+        const newB = Math.min(255, 0.272 * r + 0.534 * g + 0.131 * b);
+
+        data[i] = r * (1 - intensity) + newR * intensity;
+        data[i + 1] = g * (1 - intensity) + newG * intensity;
+        data[i + 2] = b * (1 - intensity) + newB * intensity;
+    }
+}
```

Comment:
We can reduce code duplication with something like this :


```suggestion
function mainLoop(data: Uint8ClampedArray, intensity: number, executeFilter): void {
    for (let i = 0; i < data.length; i += 4) {
        const r = data[i];
        const g = data[i + 1];
        const b = data[i + 2];
        
        executeFilter({r,g,b})
    }
}

function applySepia(data: Uint8ClampedArray, intensity: number): void {
 const newR = Math.min(255, 0.393 * r + 0.769 * g + 0.189 * b);
        const newG = Math.min(255, 0.349 * r + 0.686 * g + 0.168 * b);
        const newB = Math.min(255, 0.272 * r + 0.534 * g + 0.131 * b);

        data[i] = r * (1 - intensity) + newR * intensity;
        data[i + 1] = g * (1 - intensity) + newG * intensity;
        data[i + 2] = b * (1 - intensity) + newB * intensity;
}
```